Dockerfile

```diff
@@ -19,15 +35,24 @@ RUN go get github.com/google/gops
 #
 # Start a new stage from scratch
 #
-FROM alpine
+FROM alpine:3.7
+RUN adduser -u 1000 -D user
+COPY --from=proot /proot /home/user/.runrootless/runrootless-proot
+COPY --from=runc /runc /home/user/bin/runc
+
 RUN apk --no-cache add ca-certificates
-WORKDIR /app
-COPY --from=builder /root/src/pulsar-heartbeat /app/
+COPY --from=builder /root/src/pulsar-heartbeat /home/user
 
 # a kesque cert but it can overwritten by mounting the same path ca-bundle.crt
 COPY --from=builder /root/config/kesque-pulsar.cert /etc/ssl/certs/ca-bundle.crt
 
 # Copy debug tools
-COPY --from=builder /go/bin/gops /app/gops
+COPY --from=builder /go/bin/gops /home/user/gops
+RUN mkdir /home/user/run
+RUN chown -R user /home/user
```

Comment:
Nit: this can lead to bloat in a docker image's size. When I inspected the layers of the docker image, I can see that this command adds about 23 MB. The image is about 49 MB. You can alternatively use `--chown` as you `COPY` files from the builder to the target image.

Reply:
Reference: https://hub.docker.com/layers/datastax/pulsar-heartbeat/v0.0.6/images/sha256-edf68d6d8cc6b7c7493e8e05ac269c220c5fb861ddbd4eb3caa5c84ee1cec7c0?context=explore. <img width="498" alt="Screen Shot 2021-06-14 at 9 44 06 PM" src="https://user-images.githubusercontent.com/47911938/121989621-a6d9b100-cd59-11eb-93b1-a24722f7f251.png">